lib/spack/spack/util/web.py

```diff
@@ -550,9 +565,35 @@ def _spider(url, collect_nested):
             page = codecs.getreader("utf-8")(response).read()
             pages[response_url] = page
 
-            # Parse out the links in the page
+            # Parse out the include-fragments in the page
+            # https://github.github.io/include-fragment-element
+            include_fragment_parser = IncludeFragmentParser()
+            include_fragment_parser.feed(page)
+
+            fragments = set()
+            while include_fragment_parser.links:
+                raw_link = include_fragment_parser.links.pop()
+                abs_link = url_util.join(response_url, raw_link.strip(), resolve_href=True)
+
+                try:
+                    # This seems to be text/html, though text/fragment+html is also used
+                    fragment_response_url, _, fragment_response = read_from_url(abs_link, "text/html")
+                except Exception as e:
+                    tty.debug("Error reading fragment: %s:%s" % (type(e), str(e)), traceback.format_exc())
```

Comment:
This is failing style checks due to the long line, but before fixing that this needs to be a narrower exception catch in any case. The catch is intended to guard against dead fragment links. It seems release pages (e.g. https://github.com/VcDevel/Vc/releases) include an `include-fragment` with `src` to `/security/overall-count` (e.g. https://github.com/VcDevel/Vc/security/overall-count), which doesn't exist. That causes the code to jump to the outer exception catch without parsing any links ever.

Reply:
```suggestion
                except Exception as e:
                    msg = f"Error reading fragment: {(type(e), str(e)}:{traceback.format_exc()}"
                    tty.debug(msg)
```

What about converting to an F-string to save space?

Reply:
Do we allow those now? I guess they are since they were introduced in python 3.6 and spack requires that as a minimum.

Reply:
We do! Exactly, now that the minimum python version has moved up their fair game as I understand it.